### ISSUE TITLE
terraform-aws-provider v3.26.0+: aws-go-sdk v1.37.0+ causes Terraform to fail reading credentials from environment if SSO configuration in ~/.aws/config is incomplete

### ISSUE THREAD

**Issue by christophetd** [hashicorp/terraform-provider-aws#17370]
See also: https://github.com/aws/aws-sdk-go/issues/3768

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

- Terraform v0.14.5
- Terraform AWS provider v3.26.0. **Confirmed this bug does not occur on v3.25.0**

### Configuration Files

Terraform configuration file:
```
terraform {
  required_providers {
    aws = {
      source = "hashicorp/aws"
      version = "3.26.0"
    }
  }
}

provider "aws" {
  region = "eu-west-1"
}

resource "aws_iam_user" "test" {
	name = "test"
}
```
`~/.aws/config`:

```
[default]
sso_start_url=https://org.awsapps.com/start
sso_region=eu-central-1
cli_pager=
region=eu-west-1

[okta]
aws_saml_url=home/amazon_aws/xxx
```

### Description

Authentication. With the configuration described above, using aws-vault / the AWS CLI, I'm authenticating using AWS SSO:

```
$ aws-vault exec dev # Or using the AWS CLI
$ aws sts get-caller-identity
{
    "UserId": "AROXXXXXXXx:christophe@domain.tld",
    "Account": "123456789123",
    "Arn": "arn:aws:sts::123456789123:assumed-role/AWSReservedSSO_MyRole_xxx/christophe@domain.tld"
}
```

**Expected behavior**: When running `terraform apply`, the credentials from my environment are used to authenticate against AWS

**Actual behavior**:

```
$ terraform apply
Error: error configuring Terraform AWS Provider: Error creating AWS session: profile "default" is configured to use SSO but is missing required configuration: sso_account_id, sso_role_name
```

Note that adding random sso_account_id and sso_role_name to `~/.aws/config` does allow to work through this bug, e.g.

```
[default]
...
sso_account_id=123
sso_role_name=foo
```

### References

https://github.com/aws/aws-sdk-go/issues/3768

**Comment by ewbankkit** [comment]
Related:

* https://github.com/hashicorp/terraform-provider-aws/issues/17353
* https://github.com/hashicorp/terraform-provider-aws/issues/17372

**Comment by christophetd** [comment]
This was confirmed in https://github.com/aws/aws-sdk-go/issues/3768 to be an aws-go-sdk issue which will be fixed in their next release.

**Comment by ewbankkit** [comment]
Now available via [AWS SDK v1.37.2](https://github.com/aws/aws-sdk-go/releases/tag/v1.37.2) which should appear as a PR in this repo shortly.
